src/visit.rs

```diff
@@ -649,4 +762,74 @@ mod test {
             .unwrap();
         assert_eq!(discovered.mutants.as_slice(), &[]);
     }
+
+    /// Helper function for `find_path_attribute` tests
+    fn run_find_path_attribute(
+        token_stream: TokenStream,
+    ) -> std::result::Result<Option<Utf8PathBuf>, String> {
+        let token_string = token_stream.to_string();
+        let item_mod = syn::parse_str::<syn::ItemMod>(&token_string).unwrap_or_else(|err| {
+            panic!("Failed to parse test case token stream: {token_string}\n{err}")
+        });
+        find_path_attribute(&item_mod.attrs)
+    }
+
+    #[test]
+    fn find_path_attribute_on_module_item() {
+        let outer = run_find_path_attribute(quote! {
+            #[path = "foo_file.rs"]
+            mod foo;
+        });
+        assert_eq!(outer, Ok(Some(Utf8PathBuf::from("foo_file.rs"))));
+
+        let inner = run_find_path_attribute(quote! {
+            mod foo {
+                #![path = "foo_folder"]
+
+                #[path = "file_for_bar.rs"]
+                mod bar;
+            }
+        });
+        assert_eq!(inner, Ok(Some(Utf8PathBuf::from("foo_folder"))));
+    }
+
+    #[test]
+    fn reject_invalid_path_attribute_on_module_item() {
+        let dots = run_find_path_attribute(quote! {
+            #[path = "contains_.._dots.rs"]
+            mod dots;
+        });
+        assert_eq!(dots, Err("contains_.._dots.rs".to_owned()));
```

Comment:
Actually, thinking about this more, rustc does accept path attributes that have multiple components or that contain `..`, so we should too.

```rust
#[path = "../c/c.rs"]
mod c;

fn a() {}
```

Sorry for the detour, can you pull that Err case out?

Reply:
Sure thing. Do we still want to check the depth, to make sure we're staying inside the cargo workspace?
I'd have to check whether cargo allows relative paths outside the workspace... but it seems like cargo-mutants shouldn't try to apply mutants outside the workspace

Reply:
Good point. It looks like rustc will follow absolute paths that point outside of the workspace, but we probably should not. Maybe we should check this at the point that we go to open the new file, making sure that it's inside the overall source directory.

Reply:
I added a test in error_value to check for the relative path warning.  The "testdata" file copy code has some duplication, since it needs to maintain the relative paths between two testdata folders.... let me know what you think.

Reply:
Also removed the `failed` flag as part of the rollback; it makes sense for discovery to attempt to continue if at all possible.